### PR TITLE
Add Patience Sort

### PR DESCRIPTION
**Pull request by JoaoVictorNascimento**
### **Describe your change:**
* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by JoaoVictorNascimento** [comment]
@mateuszz0000 could you review my pull request?

File: sorts/patience_sort.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+from bisect import bisect_left
+from functools import total_ordering
+from heapq import merge
+
+"""
```
Comment (mateuszz0000) [inline sorts/patience_sort.py:5]: Please provide some url for theory

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]

File: sorts/patience_sort.py (hunk 1 above)
Comment (JoaoVictorNascimento) [reply sorts/patience_sort.py:5]: Added

**Review by mateuszz0000 — APPROVED** [review]
Nice! Well done :)

**Comment by JoaoVictorNascimento** [comment]
Thanks @mateuszz0000! Could you put the hacktoberfest-accepted label?

**Pull request merged**